packages/react-checkbox/src/components/Checkbox/useCheckbox.tsx

```diff
@@ -1,101 +1,94 @@
 import * as React from 'react';
 import {
-  getNativeElementProps,
+  getPrimarySlotNativeProps,
+  getRootSlotNativeProps,
   resolveShorthand,
   useControllableState,
+  useEventCallback,
   useId,
   useIsomorphicLayoutEffect,
   useMergedRefs,
-  useEventCallback,
 } from '@fluentui/react-utilities';
-import { CheckboxProps, CheckboxSlots, CheckboxState } from './Checkbox.types';
+import { CheckboxProps, CheckboxState } from './Checkbox.types';
 import { Mixed12Regular, Mixed16Regular, Checkmark12Regular, Checkmark16Regular } from './DefaultIcons';
 import { Label, LabelProps } from '@fluentui/react-label';
 
-/**
- * Array of all shorthand properties listed as the keys of CheckboxSlots
- */
-export const checkboxShorthandProps: Array<keyof CheckboxSlots> = ['root', 'indicator', 'input'];
-
 /**
  * Create the state required to render Checkbox.
  *
  * The returned state can be modified with hooks such as useCheckboxStyles,
  * before being passed to renderCheckbox.
  *
  * @param props - props from this instance of Checkbox
- * @param ref - reference to root HTMLElement of Checkbox
+ * @param ref - reference to <input> element of Checkbox
  */
-export const useCheckbox = (props: CheckboxProps, ref: React.Ref<HTMLElement>): CheckboxState => {
-  const { disabled = false, circular = false, required = false, id, rootId, onChange: userOnChange } = props;
+export const useCheckbox = (props: CheckboxProps, ref: React.Ref<HTMLInputElement>): CheckboxState => {
+  const { children, circular, size = 'medium', labelPosition = 'after' } = props;
+
   const [checked, setChecked] = useControllableState({
     defaultState: props.defaultChecked,
     state: props.checked,
     initialState: false,
   });
 
-  const inputInternalRef = React.useRef<HTMLInputElement>(null);
-  const inputShorthand = resolveShorthand(props.input, {
-    required: true,
-    defaultProps: {
-      disabled,
-      type: 'checkbox',
-      required,
-    },
-  });
-
   const state: CheckboxState = {
+    children,
     circular,
     checked,
-    size: 'medium',
-    labelPosition: 'after',
-    rootId,
+    size,
+    labelPosition,
     components: {
       root: props.children !== undefined ? (Label as React.ComponentType<LabelProps>) : 'span',
       indicator: 'div',
       input: 'input',
     },
-    input: {
-      ...inputShorthand,
-      ref: useMergedRefs(inputShorthand.ref, inputInternalRef),
-    },
+    root: resolveShorthand(props.root, {
+      required: true,
+      defaultProps: getRootSlotNativeProps(props),
+    }),
+    input: resolveShorthand(props.input, {
+      required: true,
+      defaultProps: {
+        type: 'checkbox',
+        id: useId('checkbox-', props.id),
+        ref,
+        // Exclude checked/defaultChecked/children from the input slot; they have custom handling in useCheckbox
+        ...getPrimarySlotNativeProps('input', props, /* excludedPropNames:*/ ['checked', 'defaultChecked', 'children']),
```

Comment:
Sure, I'll make that change. I was originally attempting to match the API of `getNativeElementProps`, since the idea is that this is a thin wrapper for that function. But there's no need to perpetuate a bad API :)